src/main/webapp/script.js

```diff
@@ -107,8 +108,19 @@ function createResultCard(name, address, photos, rating, totalCardCount) {
 
   const websiteButton = document.createElement('button');
   websiteButton.className = 'results-website-button';
-  websiteButton.innerText = 'Visit Website';
-  // TODO: Add website link
+  if (website.includes('maps.google.com')) {
+    websiteButton.innerText = 'Visit Website on Google Maps';
+    linkWebsite(website, websiteButton);
+  }
+  else if (website === '') {
```

Comment:
This might be a case we need to address further, I'm not sure having just a gray rectangle is the best solution but this works for now as this will take me doing some frontend-magic to make it change. 

Will make an issue to track, though will this case be very prevalent as all of our results are coming from Google Maps and theoretically should have a listing?

Reply:
Theoretically, this case should never happen since the places API should always at least return their reference on maps. I could only foresee this happening if the entire API call fails, and then all of the websites would be unavailable.

Reply:
Just chiming in to commend the good thought process here. That being said, I think this workaround also works fine.